custom_components/tryfi/pytryfi/fiPet.py

```diff
@@ -1,136 +1,494 @@
         self._petId = petId
-        self._name = None
-        self._homeCityState = None
-        self._yearOfBirth = None
-        self._monthOfBirth = None
-        self._gender = None
-        self._currPlaceName = None
-        self._currPlaceAddress = None
-        self._device = None
-        self._weight = None
-        self._lastUpdated = None
-        self._locationLastUpdate = None
 
     def setPetDetailsJSON(self, petJSON: dict):
         self._name = petJSON.get('name')
         self._homeCityState = petJSON.get('homeCityState')
-        self._yearOfBirth = petJSON['yearOfBirth']
-        self._monthOfBirth = int(petJSON['monthOfBirth'])
-        self._dayOfBirth = int(petJSON['dayOfBirth'])
+        try:
+            self._yearOfBirth = int(petJSON['yearOfBirth'])
+        except Exception:
+            LOGGER.warning("Unknown Year of Birth")
+            self._yearOfBirth = 1900
+        try:
+            self._monthOfBirth = int(petJSON['monthOfBirth'])
+        except Exception:
+            LOGGER.warning("Unknown Month of Birth")
+            self._monthOfBirth = 1
+        try:
+            self._dayOfBirth = int(petJSON['dayOfBirth'])
+        except Exception:
+            LOGGER.warning("Unknown day of birth")
+            self._dayOfBirth = None
         self._gender = petJSON.get('gender')
         self._weight = float(petJSON['weight']) if 'weight' in petJSON else None
-        self._breed = petJSON['breed'].get('name') if 'breed' in petJSON else None
+        try:
+            self._breed = petJSON['breed']['name']
+        except Exception:
+            LOGGER.warning("Unknown Breed of Dog")
+            self._breed = None
+            #track last updated
+        self._lastUpdated = datetime.datetime.now()
         try:
             self._photoLink = petJSON['photos']['first']['image']['fullSize']
         except Exception:
             #capture_exception(e)
             LOGGER.warning("Cannot find photo of your pet. Defaulting to empty string.")
             self._photoLink = ""
-        self._device = FiDevice(petJSON['device']['id'])
-        self._device.setDeviceDetailsJSON(petJSON['device'])
-        self._lastUpdated = datetime.datetime.now()
+        # Check if pet has a device
+        if petJSON.get('device'):
+            self._device = FiDevice(petJSON['device']['id'])
+            self._device.setDeviceDetailsJSON(petJSON['device'])
+            self._connectedTo = self.setConnectedTo(petJSON['device']['lastConnectionState'])
+        else:
+            self._device = None
+            self._connectedTo = None
+            LOGGER.info(f"Pet {self._name} has no device assigned")
+
+        # Initialize behavior metrics to 0 (will be updated later if available)
+        self._dailyBarkingCount = 0
+        self._dailyBarkingDuration = 0
+        self._weeklyBarkingCount = 0
+        self._weeklyBarkingDuration = 0
+        self._monthlyBarkingCount = 0
+        self._monthlyBarkingDuration = 0
```

Comment:
Optional, but one thing I was considering is just have a StatsObject and the fiPet has just a "_dailyStats", "_monthlyStats", "_weeklyStats", so we don't have to duplicate this code.